tests/components/geo_location/test_demo.py

```diff
@@ -0,0 +1,87 @@
+"""The tests for the demo platform."""
+import unittest
+
+from homeassistant.components import geo_location
+from homeassistant.components.geo_location.demo import DemoManager, \
+    NUMBER_OF_DEMO_DEVICES, setup_platform
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import callback
+from homeassistant.setup import setup_component
+from tests.common import get_test_home_assistant, assert_setup_component
+
+CONFIG = {
+    geo_location.DOMAIN: [
+        {
+            'platform': 'demo'
+        }
+    ]
+}
+
+
+class TestDemoPlatform(unittest.TestCase):
+    """Test the demo platform."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_setup_via_config(self):
+        """Test setup of demo platform via configuration."""
+        with assert_setup_component(1, geo_location.DOMAIN):
+            self.assertTrue(setup_component(self.hass, geo_location.DOMAIN,
+                                            CONFIG))
+
+    def test_setup_platform(self):
+        """Test setup of demo platform."""
+        devices = []
+
+        @callback
+        def add_devices_callback(events):
+            """Add recorded devices."""
+            devices.extend(events)
+
+        self.assertTrue(setup_platform(self.hass, None, add_devices_callback,
```

Comment:
Please set up the platform by setting up the component with config, as above, then assert that the correct states are populated in the state machine.

Reply:
OK